common/CMakeLists.txt

```diff
@@ -47,6 +47,16 @@ if (BUILD_SHARED_LIBS)
     set_target_properties(${TARGET} PROPERTIES POSITION_INDEPENDENT_CODE ON)
 endif()
 
+# Check for curl
+find_package(CURL QUIET)
```

Comment:
Please correct if I’m wrong, but here we check if curl is found on build system, then we turn on LLAMA_CURL. I wonder in case user have libcurl but don’t want to enable this feature in llama.cpp (to make sure there’s no depenency for example), then can we do that ?

Reply:
AFAIK It's dynamic linking so it will trigger linking to `libcurl` at runtime only if user use `--model-url` or `llama_load_model_from_url` and at this stage I think it's OK. If users complain that  `ldd` is listing `libcurl.so` we can later on introduce `LLAMA_CURL=ON` cmake option as in the make build toolchain.

Reply:
At least on Linux that’s not the case. If the libcurl.so is listed in ldd, than it will be dynamically linked the moment we run the binary (even if the function is not used). In fact, for this reason, I avoid having dynamic link when possible (sometimes it breaks the portablity due to mismatch version of library in build time & runtime). Also I think in Makefile we only build with libcurl only if specified, so should we match the behavior of cmake & makefile?